Lib/test/test_netrc.py

```diff
@@ -92,221 +196,258 @@ def test_invalid_tokens(self):
+        self._test_comment(make_nrc, """\
             machine foo.domain.com login bar password #pass #comment
             machine bar.domain.com login foo password pass
             """, '#pass')
 
+    @unittest.skipUnless(os.name == 'posix', 'POSIX only test')
+    @unittest.skipIf(pwd is None, 'security check requires pwd module')
+    @support.os_helper.skip_unless_working_chmod
+    def test_non_anonymous_security(self):
+        # This test is incomplete since we are normally not run as root and
+        # therefore can't test the file ownership being wrong.
+        content = """
+            machine foo.domain.com login bar password pass
+            default login foo password pass
+        """
+        mode = 0o662
+
+        # Use ~/.netrc and login is not anon
+        with self.assertRaises(netrc.NetrcParseError):
+            NetrcBuilder.use_default_netrc_in_home(content, mode=mode)
+
+        # Don't use default file
+        nrc = NetrcBuilder.use_file_argument(content, mode=mode)
+        self.assertEqual(nrc.hosts['foo.domain.com'],
+                            ('bar', '', 'pass'))
 
     @unittest.skipUnless(os.name == 'posix', 'POSIX only test')
     @unittest.skipIf(pwd is None, 'security check requires pwd module')
-    @os_helper.skip_unless_working_chmod
-    def test_security(self):
+    @support.os_helper.skip_unless_working_chmod
+    @support.subTests('make_nrc', ALL_NETRC_FILE_SCENARIOS)
+    def test_anonymous_security(self, make_nrc):
         # This test is incomplete since we are normally not run as root and
         # therefore can't test the file ownership being wrong.
-        d = os_helper.TESTFN
-        os.mkdir(d)
-        self.addCleanup(os_helper.rmtree, d)
-        fn = os.path.join(d, '.netrc')
-        with open(fn, 'wt') as f:
-            f.write("""\
-                machine foo.domain.com login bar password pass
-                default login foo password pass
-                """)
-        with os_helper.EnvironmentVarGuard() as environ:
-            environ.set('HOME', d)
-            os.chmod(fn, 0o600)
-            nrc = netrc.netrc()
-            self.assertEqual(nrc.hosts['foo.domain.com'],
-                             ('bar', '', 'pass'))
-            os.chmod(fn, 0o622)
-            self.assertRaises(netrc.NetrcParseError, netrc.netrc)
-        with open(fn, 'wt') as f:
-            f.write("""\
-                machine foo.domain.com login anonymous password pass
-                default login foo password pass
-                """)
-        with os_helper.EnvironmentVarGuard() as environ:
-            environ.set('HOME', d)
-            os.chmod(fn, 0o600)
-            nrc = netrc.netrc()
-            self.assertEqual(nrc.hosts['foo.domain.com'],
-                             ('anonymous', '', 'pass'))
-            os.chmod(fn, 0o622)
+        content = """\
+            machine foo.domain.com login anonymous password pass
```

Comment:
Keep the old test; we'll add an other test in a follow-up PR for anonymous entry without a default.